tcp/client/session.go

```diff
@@ -165,123 +149,96 @@ func (s *Session) Close() error {
 }
 
 func (s *Session) Sequence() uint64 {
-	return atomic.AddUint64(&s.sequence, 1)
+	return s.sequence.Inc()
 }
 
 func (s *Session) Context() context.Context {
-	return *s.ctx.Load().(*context.Context)
+	return *s.ctx.Load().(*context.Context) //nolint:forcetypeassert
```

Comment:
This seems like a data race, in `SetContextValue` above `s.ctx` is protected by the `s.mutex`. But the read here is not protected.

Reply:
No it's not data race:
This mutex is used when two goroutines call `SetContextValue `at the same time. Regardless of order, we want both values to be stored in the context with different keys.  Due to the fact that the operation consists of two steps, the connection can only contain one of the values without the mutex.

The operation `s.ctx.Load()` is atomic so it is goroutine safe